demo/laneLoader.js

```diff
@@ -249,14 +249,19 @@ async function createLaneGeometries (lanes, supplierNum, annotationMode, volumes
         geometrySpine.vertices.push(new THREE.Vector3(spine.x(), spine.y(), spine.z()));
       }
 
-      // Spine points use the closest left lane point to determine validity
+      // Spine points use the closest left lane or right lane point to determine validity
       if (window.usingInvalidLanesSchema) {
         const spinePoint = new THREE.Vector3(spine.x(), spine.y(), spine.z());
+
         const closestLeftIndex = geometryLeft.vertices.reduce((acc, current, i) => {
           return current.distanceTo(spinePoint) < geometryLeft.vertices[acc].distanceTo(spinePoint) ? i : acc;
         }, 0);
 
-        spineValidities.push(leftValidities[closestLeftIndex])
+        const closestRightIndex = geometryRight.vertices.reduce((acc, current, i) => {
+          return current.distanceTo(spinePoint) < geometryRight.vertices[acc].distanceTo(spinePoint) ? i : acc;
+        }, 0);
+
+        spineValidities.push(leftValidities[closestLeftIndex] || rightValidities[closestRightIndex]);
```

Comment:
Shouldn't it be `leftValidities[closestLeftIndex] || rightValidities[closestRightIndex` ?